### ISSUE TITLE
Understanding accumulate_grad_batches  parameter? 

### ISSUE THREAD

**Issue by shamanez** [PyTorchLightning/pytorch-lightning#5636]
I am very new to PL. As far as I understand **accumulate_grad_batches** works similar to  **'gradient_accumulation_steps'** , where the main task is to increase the effective batch size.  But I do not see any change in training epoch step count when increasing the  **accumulate_grad_batches** parameters.

Let's say, I have a dataset of 1000 examples and my batch_size is one and I only use a single GPU. So in this case, if I use the value 2  for the **accumulate_grad_batches**,  the number of steps for an epoch should be shown as 500 (logger). But I still see 1000.

Is it a bug or PL doesn't divide the number of steps when showing in the log?

**Comment by rohitgr7** [comment]
total step count will remain the same since it refers to total batches, but internally `optimizer/scheduler.step` is updated accordingly. You can check `self.global_step` with and without accumulation.

**Comment by tchaton** [comment]
Hey @shamanez,

I believe @rohitgr7 properly answered your question.
I will close this issue. Feel free to re-open it if something is missing.

Best,
T.C

**Comment by shamanez** [comment]
Thanks a lot.sorted!
